permissions/src/androidMain/kotlin/dev/icerock/moko/permissions/PermissionsControllerImpl.kt

```diff
@@ -194,12 +177,20 @@ class PermissionsControllerImpl(
         applicationContext.startActivity(intent)
     }
 
+    private fun unbindActivity() {
+        launcherHolder.value?.unregister()
+        activityHolder.value = null
+        launcherHolder.value = null
+    }
+
     private companion object {
         private const val AWAIT_ACTIVITY_TIMEOUT_DURATION_MS = 2000L
     }
 }
 
-private class PermissionCallback(
-    val permission: Permission,
-    val callback: (Result<Unit>) -> Unit
-)
+private sealed class PermissionRequestResult {
+    object Granted : PermissionRequestResult()
+    object Denied : PermissionRequestResult()
+    object DeniedAlways : PermissionRequestResult()
+    object Cancelled : PermissionRequestResult()
+}
```

Comment:
this should be simple enum, no need to use sealed here

Reply:
I agree. Will make the change